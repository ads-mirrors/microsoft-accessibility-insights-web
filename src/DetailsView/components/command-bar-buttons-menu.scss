// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

.command-bar-buttons-menu-button {
    font-size: 16px;
    padding: 9px !important;
    color: $primary-text !important;

    &:hover {
        background-color: $neutral-2 !important;
    }

    &:focus {
        outline: 1px solid;
    }
}

.menu-list {
    top: 19px !important;
}

.menu-item {
    padding-bottom: unset !important;

    &:hover {
        background: none !important;
    }
}

.menu-button {
    align-items: flex-start !important;
    font-weight: 400 !important;
    margin-left: -8px !important;

    &:hover {
        background: 'none !important';
        color: $insights-button-hover;

        & > span {
            & > svg {
                color: $insights-button-hover;
            }
        }
    }
}

.refresh-icon {
    color: $insights-button-hover !important;
}

.command-bar-buttons-submenu {
    min-width: fit-content;

    button {
        height: 36px;
    }
}

.command-bar-buttons-menu {
    height: 17px;
    margin-left: 9px !important;
    margin-top: 8px !important;
    margin-bottom: 16px !important;
}
