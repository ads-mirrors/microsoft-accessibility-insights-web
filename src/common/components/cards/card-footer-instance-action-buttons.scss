// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../styles/colors.scss';

.kebab-menu-icon {
    flex-grow: 1; // Center the custom SVG
    padding-left: 9px !important;
    padding-right: 9px !important;
    padding-top: 6px !important;
    padding-bottom: 6px !important;
    @media screen and (forced-colors: active) {
        color: buttontext !important;
    }
}

.fileissue {
    font-weight: 400 !important;

    &:hover {
        & > span {
            & > svg {
                color: $insights-button-hover;
            }
        }
    }

    &:focus {
        outline: 1px solid;
    }
}

.copyfailuredetails {
    font-weight: 400 !important;

    &:hover {
        & > span {
            & > svg {
                color: $insights-button-hover;
            }
        }
    }
}

.menu-Button {
    &:hover {
        border: 1px solid $primary-text !important;
        color: $primary-text !important;
    }

    &:focus {
        outline: 1px solid;
    }
}

.kebab-menu-button {
    margin-right: 8px;

    // Since the menu uses a custom SVG, set the button shape:
    width: 32px;
    height: 32px;

    &:hover {
        background-color: $menu-item-background-hover;
    }

    &:global(.is-expanded) {
        background-color: $menu-item-background-active;
    }

    // Add hover and active state:
    &:global(.is-expanded) .kebab-menu-icon,
    &:hover .kebab-menu-icon {
        @media screen and (forced-colors: active) {
            color: highlight;
        }
    }
}
