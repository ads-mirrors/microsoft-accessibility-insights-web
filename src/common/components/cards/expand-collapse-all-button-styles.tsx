// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { makeStyles, tokens } from '@fluentui/react-components';

export const useExpandCollapseAllButtonStyles: any = makeStyles({
    expandCollapseAllButton: {
        display: 'flex',
        padding: 'unset',
        paddingRight: '27px!important',
        paddingLeft: '1px!important',
        marginLeft: '1px!important',
        marginTop: '2px!important',
        fontWeight: tokens?.fontSizeBase400,
    },

    customStyleIcon: {
        color: tokens.colorNeutralForeground2BrandHover,
    },
});
