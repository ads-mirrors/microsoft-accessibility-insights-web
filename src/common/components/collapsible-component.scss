// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

span.collapsible-title {
    padding-left: 0.5vw;
}

.collapsible {
    cursor: pointer;
    width: auto;
    border: none;
    outline: none;
    line-height: 0.25px;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    padding-top: 5px;
    padding-bottom: 5px;
    padding-right: 27px !important;
    padding-left: 1px !important;

    h2 {
        font-weight: $normal-title-font-weight;
        font-size: 17px;
    }
}

.collapsible-content {
    overflow: hidden;
    padding-left: calc(0.5vw + 13px); // 13px is Chevron width.
}
