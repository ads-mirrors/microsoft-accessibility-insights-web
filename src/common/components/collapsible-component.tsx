// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Button } from '@fluentui/react-components';
import { css } from '@fluentui/utilities';
import styles from 'common/components/collapsible-component.scss';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import * as React from 'react';

export interface CollapsibleComponentProps {
    header: JSX.Element;
    content: JSX.Element;
    contentClassName?: string;
    containerClassName?: string;
}

interface CollapsibleComponentState {
    showContent: boolean;
}

export class CollapsibleComponent extends React.Component<
    CollapsibleComponentProps,
    CollapsibleComponentState
> {
    constructor(props: CollapsibleComponentProps) {
        super(props);
        this.state = { showContent: true };
    }

    private onClick = (): void => {
        this.setState(prevState => ({ showContent: !prevState.showContent }));
    };

    public render(): JSX.Element {
        const showContent = this.state.showContent;
        let content: JSX.Element | null = null;

        if (showContent) {
            content = (
                <div className={css(this.props.contentClassName, styles.collapsibleContent)}>
                    {this.props.content}
                </div>
            );
        }

        return (
            <div className={css(this.props.containerClassName)}>
                <Button
                    className={styles.collapsible}
                    appearance="transparent"
                    onClick={this.onClick}
                    aria-expanded={showContent}
                >
                    {showContent ? (
                        <FluentUIV9Icon iconName="ChevronDown24Regular" />
                    ) : (
                        <FluentUIV9Icon iconName="ChevronRight24Regular" />
                    )}
                    {this.props.header}
                </Button>
                {content}
            </div>
        );
    }
}
