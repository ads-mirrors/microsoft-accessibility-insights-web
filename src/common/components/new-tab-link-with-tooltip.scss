// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';

:global(.high-contrast-theme) {
    .insights-tooltip {
        outline-style: solid;
        outline-width: 1px;
        outline-color: $always-white;
    }
}

.insights-tooltip-host {
    display: inline-block;
    min-width: 0;
}

.insights-link {
    display: flex;

    &:hover,
    &:active {
        text-decoration: underline !important;
    }
}

.insights-link-icon {
    display: flex;

    &:hover,
    &:active {
        text-decoration: none !important;
    }
}

@media screen and (forced-colors: active) {
    .insights-link:hover,
    .insights-link-icon:hover {
        forced-color-adjust: none;
        color: ButtonText !important;
    }
}
