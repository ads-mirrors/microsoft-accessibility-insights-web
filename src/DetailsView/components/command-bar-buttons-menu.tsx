// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    Menu,
    MenuButton,
    MenuList,
    MenuPopover,
    MenuProps,
    MenuTrigger,
    Tooltip,
} from '@fluentui/react-components';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import { NamedFC } from 'common/react/named-fc';
import { ButtonRefFunction } from 'DetailsView/components/details-view-command-bar';
import { StartOverMenuItem } from 'DetailsView/components/start-over-component-factory';
import * as React from 'react';
import styles from './command-bar-buttons-menu.scss';

export type CommandBarButtonsMenuProps = {
    renderExportReportButton: () => JSX.Element | null;
    saveAssessmentButton?: JSX.Element | null;
    loadAssessmentButton?: JSX.Element | null;
    transferToAssessmentButton?: JSX.Element | null;
    getStartOverMenuItem: () => StartOverMenuItem;
    buttonRef?: ButtonRefFunction;
    hasSubMenu?: boolean;
};

export const CommandBarButtonsMenu = NamedFC<CommandBarButtonsMenuProps>(
    'CommandBarButtonsMenu',
    props => {
        const [open, setOpen] = React.useState(false);
        const onOpenChange: MenuProps['onOpenChange'] = (e, data) => setOpen(data.open);
        const exportButton = props.renderExportReportButton();

        const overflowItems: any[] = [];
        if (exportButton != null) {
            overflowItems.push({
                key: 'export report',
                children: <>{exportButton}</>,
            });
        }
        if (props.saveAssessmentButton && props.loadAssessmentButton) {
            overflowItems.push(
                {
                    key: 'save assessment',
                    children: <>{props.saveAssessmentButton}</>,
                },
                {
                    key: 'load assessment',
                    children: <>{props.loadAssessmentButton}</>,
                },
            );
        }

        if (props.transferToAssessmentButton) {
            overflowItems.push({
                key: 'transfer to assessment',
                children: <>{props.transferToAssessmentButton}</>,
            });
        }

        overflowItems.push({
            key: 'start over',
            ...props.getStartOverMenuItem(),
        });

        return (
            <>
                <Menu open={open} onOpenChange={onOpenChange}>
                    <Tooltip content="More actions" relationship="label">
                        <MenuTrigger disableButtonEnhancement>
                            <MenuButton
                                appearance="transparent"
                                aria-label="More actions"
                                icon={<FluentUIV9Icon iconName="MoreHorizontalRegular" />}
                                className={styles.commandBarButtonsMenuButton}
                                ref={() => props.buttonRef}
                            />
                        </MenuTrigger>
                    </Tooltip>
                    <MenuPopover
                        style={{
                            border: 'unset !important',
                            borderRadius: 'unset !important',
                        }}
                    >
                        <MenuList className={styles.menuList}>
                            {overflowItems.map((item, index) => (
                                <span key={index}>{item?.children}</span>
                            ))}
                        </MenuList>
                    </MenuPopover>
                </Menu>
            </>
        );
    },
);
