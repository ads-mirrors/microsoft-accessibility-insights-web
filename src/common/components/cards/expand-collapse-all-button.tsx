// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Button } from '@fluentui/react-components';
import { useExpandCollapseAllButtonStyles } from 'common/components/cards/expand-collapse-all-button-styles';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import { CardSelectionMessageCreator } from 'common/message-creators/card-selection-message-creator';
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';

export type ExpandCollapseAllButtonProps = {
    allCardsCollapsed: boolean;
    cardSelectionMessageCreator: CardSelectionMessageCreator;
};

export const ExpandCollapseAllButton = NamedFC<ExpandCollapseAllButtonProps>(
    'ExpandCollapseAllButton',
    props => {
        const getStyles: any = useExpandCollapseAllButtonStyles();
        const { allCardsCollapsed, cardSelectionMessageCreator } = props;

        let expandCollapseAllButtonHandler = cardSelectionMessageCreator.collapseAllRules;
        let buttonText = 'Collapse all';
        let ariaLabel: string | undefined = undefined;

        if (allCardsCollapsed) {
            expandCollapseAllButtonHandler = cardSelectionMessageCreator.expandAllRules;
            buttonText = 'Expand all';
            ariaLabel = 'Expand all rules to show failed instances.';
        }

        const IconName = () => {
            if (allCardsCollapsed) {
                return (
                    <FluentUIV9Icon
                        iconName="ChevronRight32Regular"
                        customClass={getStyles?.customStyleIcon}
                    />
                );
            }
            return (
                <FluentUIV9Icon
                    iconName="ChevronDown32Regular"
                    customClass={getStyles?.customStyleIcon}
                />
            );
        };

        return (
            <Button
                icon={<IconName />}
                size="medium"
                appearance="transparent"
                className={getStyles?.expandCollapseAllButton}
                aria-label={ariaLabel}
                aria-expanded={!allCardsCollapsed}
                onClick={expandCollapseAllButtonHandler}
            >
                {buttonText}
            </Button>
        );
    },
);
