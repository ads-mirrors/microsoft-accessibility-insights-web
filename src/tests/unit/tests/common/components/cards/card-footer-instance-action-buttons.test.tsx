// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    Button,
    Menu,
    MenuButton,
    MenuItem,
    MenuList,
    MenuPopover,
    MenuTrigger,
    Tooltip,
} from '@fluentui/react-components';
import { render, RenderResult } from '@testing-library/react';
import {
    CardFooterInstanceActionButtons,
    CardFooterInstanceActionButtonsDeps,
    CardFooterInstanceActionButtonsProps,
} from 'common/components/cards/card-footer-instance-action-buttons';
import {
    CardFooterMenuItem,
    CardFooterMenuItemsBuilder,
    CardFooterMenuItemsProps,
} from 'common/components/cards/card-footer-menu-items-builder';
import { allCardInteractionsSupported } from 'common/components/cards/card-interaction-support';
import { CardsViewController } from 'common/components/cards/cards-view-controller';
import { CreateIssueDetailsTextData } from 'common/types/create-issue-details-text-data';
import { UserConfigurationStoreData } from 'common/types/store-data/user-configuration-store';
import { NarrowModeStatus } from 'DetailsView/components/narrow-mode-detector';
import * as React from 'react';
import { IMock, It, Mock, Times } from 'typemoq';
import { Toast } from '../../../../../../common/components/toast';
import {
    expectMockedComponentPropsToMatchSnapshots,
    mockReactComponents,
    useOriginalReactElements,
} from '../../../../mock-helpers/mock-module-helpers';

jest.mock('@fluentui/react-components');
jest.mock('../../../../../../common/components/toast');

jest.mock('common/icons/fluentui-v9-icons');
describe('CardFooterInstanceActionButtons', () => {
    mockReactComponents([
        Toast,
        Button,
        MenuItem,
        Menu,
        MenuTrigger,
        MenuPopover,
        MenuList,
        Tooltip,
    ]);

    let defaultProps: CardFooterInstanceActionButtonsProps;
    let defaultDeps: CardFooterInstanceActionButtonsDeps;
    let menuItemsBuilderMock: IMock<CardFooterMenuItemsBuilder>;
    let cardsViewControllerMock: IMock<CardsViewController>;

    const menuItems: CardFooterMenuItem[] = [
        {
            key: 'item1',
            text: 'item 1',
            onClick: () => null,
        },
        {
            key: 'item2',
            text: 'item 2',
            onClick: () => null,
        },
    ];

    beforeEach(() => {
        menuItemsBuilderMock = Mock.ofType<CardFooterMenuItemsBuilder>();
        cardsViewControllerMock = Mock.ofType<CardsViewController>();
        defaultDeps = {
            cardInteractionSupport: allCardInteractionsSupported,
            cardFooterMenuItemsBuilder: menuItemsBuilderMock.object,
            cardsViewController: cardsViewControllerMock.object,
        } as CardFooterInstanceActionButtonsDeps;

        defaultProps = {
            deps: defaultDeps,
            userConfigurationStoreData: {} as UserConfigurationStoreData,
            issueDetailsData: {} as CreateIssueDetailsTextData,
        } as CardFooterInstanceActionButtonsProps;
    });

    it('renders as null with no menu items', () => {
        setupGetMenuItems([], defaultProps);

        const renderResult = render(<CardFooterInstanceActionButtons {...defaultProps} />);
        expect(renderResult.container.firstChild).toBeNull();
    });

    it('renders without copyFailureDetails supported', () => {
        const props = {
            ...defaultProps,
            deps: {
                ...defaultDeps,
                cardInteractionSupport: {
                    ...allCardInteractionsSupported,
                    supportsCopyFailureDetails: false,
                },
            },
        };
        setupGetMenuItems(menuItems, props);

        const renderResult = render(<CardFooterInstanceActionButtons {...props} />);

        expect(renderResult.asFragment()).toMatchSnapshot('component snapshot');
    });

    describe.each([true, false])('with isCardFooterCollapsed=%s', isCardFooterCollapsed => {
        beforeEach(() => {
            defaultProps.narrowModeStatus = { isCardFooterCollapsed } as NarrowModeStatus;
        });

        it('renders per snapshot', () => {
            setupGetMenuItems(menuItems, defaultProps);

            const renderResult = render(<CardFooterInstanceActionButtons {...defaultProps} />);

            verifySnapshots(renderResult);
        });

        function verifySnapshots(renderedElement: RenderResult): void {
            useOriginalReactElements('@fluentui/react-components', [
                'Tooltip',
                'Menu',
                'MenuTrigger',
                'MenuButton',
                'MenuPopover',
                'MenuList',
                'MenuItem',
            ]);
            expect(renderedElement.asFragment()).toMatchSnapshot('component snapshot');
            if (isCardFooterCollapsed) {
                expect(renderedElement.asFragment()).toMatchSnapshot('kebab button menu');
            }
        }
    });

    it.each(['test-kebabmenuarialabel', undefined])(
        'renders per snapshot with %s aria label passed as prop',
        ariaLabel => {
            defaultProps.narrowModeStatus = { isCardFooterCollapsed: true } as NarrowModeStatus;

            const newProps = {
                ...defaultProps,
                kebabMenuAriaLabel: ariaLabel,
            };
            setupGetMenuItems(menuItems, newProps);

            const renderResult = render(<CardFooterInstanceActionButtons {...newProps} />);

            expect(renderResult.asFragment()).toMatchSnapshot('component snapshot');
            expectMockedComponentPropsToMatchSnapshots([MenuButton], 'action button menu props');
        },
    );

    it('onIssueFilingSettingsDialogDismissed focus kebab button', async () => {
        useOriginalReactElements('@fluentui/react-components', ['Button']);
        defaultProps.narrowModeStatus = { isCardFooterCollapsed: true } as NarrowModeStatus;
        let menuItemsProps: CardFooterMenuItemsProps;

        menuItemsBuilderMock
            .setup(m =>
                m.getCardFooterMenuItems(
                    It.isObjectWith(defaultProps as unknown as CardFooterMenuItemsProps),
                ),
            )
            .returns((props: CardFooterMenuItemsProps) => {
                menuItemsProps = props;
                return menuItems;
            });
        const result = render(<CardFooterInstanceActionButtons {...defaultProps} />);
        menuItemsProps.onIssueFilingSettingsDialogDismissed();
        jest.spyOn(menuItemsProps, 'onIssueFilingSettingsDialogDismissed');
        const getButton = result.getByRole('button');

        expect(document.activeElement).toBe(getButton);
    });

    it('File issue button is focused when issue filing settings dialog is dismissed', async () => {
        defaultProps.narrowModeStatus = { isCardFooterCollapsed: false } as NarrowModeStatus;
        const fileIssueButtonMock = Mock.ofType<HTMLButtonElement>();

        let menuItemsProps: CardFooterMenuItemsProps;
        menuItemsBuilderMock
            .setup(m =>
                m.getCardFooterMenuItems(
                    It.isObjectWith(defaultProps as unknown as CardFooterMenuItemsProps),
                ),
            )
            .returns(props => {
                menuItemsProps = props;
                return menuItems;
            });

        render(
            <CardFooterInstanceActionButtons
                {...defaultProps}
                deps={{ ...defaultDeps, cardInteractionSupport: allCardInteractionsSupported }}
            />,
        );

        // call ref callback to set rendered component's ref to our button mock
        const buttonRefCallback = menuItemsProps.fileIssueButtonRef as any;
        buttonRefCallback(fileIssueButtonMock.object);

        menuItemsProps.onIssueFilingSettingsDialogDismissed();

        fileIssueButtonMock.verify(f => f.focus(), Times.once());
    });

    function setupGetMenuItems(
        menuItems: CardFooterMenuItem[],
        expectedProps: CardFooterInstanceActionButtonsProps,
    ): void {
        menuItemsBuilderMock
            .setup(m =>
                m.getCardFooterMenuItems(
                    It.isObjectWith(expectedProps as unknown as CardFooterMenuItemsProps),
                ),
            )
            .returns(() => menuItems);
    }
});
