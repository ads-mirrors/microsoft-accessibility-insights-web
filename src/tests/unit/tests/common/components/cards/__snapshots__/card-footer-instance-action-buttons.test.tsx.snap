// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooterInstanceActionButtons renders per snapshot with test-kebabmenuarialabel aria label passed as prop: action button menu props 1`] = `
{
  "appearance": "transparent",
  "aria-disabled": undefined,
  "aria-expanded": undefined,
  "aria-haspopup": "menu",
  "aria-label": "test-kebabmenuarialabel",
  "className": "menuButton",
  "data-tabster": "{"restorer":{"type":1}}",
  "icon": <FluentUIV9Icon
    iconName="MoreVerticalRegular"
  />,
  "id": "menur0",
  "onClick": [Function],
  "onContextMenu": [Function],
  "onKeyDown": [Function],
  "onKeyUp": [Function],
  "onMouseEnter": [Function],
  "onMouseLeave": [Function],
  "onMouseMove": [Function],
  "role": "button",
  "shape": "square",
  "tabIndex": 0,
}
`;

exports[`CardFooterInstanceActionButtons renders per snapshot with test-kebabmenuarialabel aria label passed as prop: component snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      aria-label="test-kebabmenuarialabel"
      class="fui-Button r1alrhcs fui-MenuButton menuButton ___14ayyxl_z6mxox0 f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 f1tme0vf f4xjyn1 f18onu3q f9ddjv3 f1fabniw fj0ryk1 f18ktai2 fwbmr0d f44c6la"
      data-tabster="{"restorer":{"type":1}}"
      id="menur0"
      role="button"
      tabindex="0"
      type="button"
    >
      <span
        class="fui-Button__icon rywnvv2 fui-MenuButton__icon"
      />
    </button>
    <mock-toast
      deps="[object Object]"
      timeoutlength="6000"
    />
  </div>
</DocumentFragment>
`;

exports[`CardFooterInstanceActionButtons renders per snapshot with undefined aria label passed as prop: action button menu props 1`] = `
{
  "appearance": "transparent",
  "aria-disabled": undefined,
  "aria-expanded": undefined,
  "aria-haspopup": "menu",
  "aria-label": undefined,
  "className": "menuButton",
  "data-tabster": "{"restorer":{"type":1}}",
  "icon": <FluentUIV9Icon
    iconName="MoreVerticalRegular"
  />,
  "id": "menur1",
  "onClick": [Function],
  "onContextMenu": [Function],
  "onKeyDown": [Function],
  "onKeyUp": [Function],
  "onMouseEnter": [Function],
  "onMouseLeave": [Function],
  "onMouseMove": [Function],
  "role": "button",
  "shape": "square",
  "tabIndex": 0,
}
`;

exports[`CardFooterInstanceActionButtons renders per snapshot with undefined aria label passed as prop: component snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      aria-expanded="false"
      aria-haspopup="menu"
      class="fui-Button r1alrhcs fui-MenuButton menuButton ___14ayyxl_z6mxox0 f1c21dwh f1p3nwhy f11589ue f1q5o8ev f1pdflbu fkfq4zb fjxutwb f1s2uweq fr80ssc f1ukrpxl fecsdlb f139oj5f ft1hn21 fuxngvv fwiml72 f1h0usnq fs4ktlq f16h9ulv fx2bmrt f1fg1p5m f1dfjoow f1j98vj9 f1tme0vf f4xjyn1 f18onu3q f9ddjv3 f1fabniw fj0ryk1 f18ktai2 fwbmr0d f44c6la"
      data-tabster="{"restorer":{"type":1}}"
      id="menur1"
      role="button"
      tabindex="0"
      type="button"
    >
      <span
        class="fui-Button__icon rywnvv2 fui-MenuButton__icon"
      />
    </button>
    <mock-toast
      deps="[object Object]"
      timeoutlength="6000"
    />
  </div>
</DocumentFragment>
`;

exports[`CardFooterInstanceActionButtons renders without copyFailureDetails supported: component snapshot 1`] = `
<DocumentFragment>
  <div>
    <mock-button
      appearance="transparent"
      as="button"
      classname="item1"
      icon="[object Object]"
      size="medium"
    >
      item 1
    </mock-button>
    <mock-button
      appearance="transparent"
      as="button"
      classname="item2"
      icon="[object Object]"
      size="medium"
    >
      item 2
    </mock-button>
  </div>
</DocumentFragment>
`;

exports[`CardFooterInstanceActionButtons with isCardFooterCollapsed=false renders per snapshot: component snapshot 1`] = `
<DocumentFragment>
  <div>
    <mock-button
      appearance="transparent"
      as="button"
      classname="item1"
      icon="[object Object]"
      size="medium"
    >
      item 1
    </mock-button>
    <mock-button
      appearance="transparent"
      as="button"
      classname="item2"
      icon="[object Object]"
      size="medium"
    >
      item 2
    </mock-button>
    <mock-toast
      deps="[object Object]"
      timeoutlength="6000"
    />
  </div>
</DocumentFragment>
`;

exports[`CardFooterInstanceActionButtons with isCardFooterCollapsed=true renders per snapshot: component snapshot 1`] = `
<DocumentFragment>
  <div>
    <mock-menu>
      <mock-menutrigger />
      <mock-menupopover
        style="border-radius: var(--borderRadiusNone);"
      >
        <mock-menulist>
          <mock-menuitem
            classname="kebabMenuIcon"
            icon="[object Object]"
            text="item 1"
          >
            item 1
          </mock-menuitem>
          <mock-menuitem
            classname="kebabMenuIcon"
            icon="[object Object]"
            text="item 2"
          >
            item 2
          </mock-menuitem>
        </mock-menulist>
      </mock-menupopover>
    </mock-menu>
    <mock-toast
      deps="[object Object]"
      timeoutlength="6000"
    />
  </div>
</DocumentFragment>
`;

exports[`CardFooterInstanceActionButtons with isCardFooterCollapsed=true renders per snapshot: kebab button menu 1`] = `
<DocumentFragment>
  <div>
    <mock-menu>
      <mock-menutrigger />
      <mock-menupopover
        style="border-radius: var(--borderRadiusNone);"
      >
        <mock-menulist>
          <mock-menuitem
            classname="kebabMenuIcon"
            icon="[object Object]"
            text="item 1"
          >
            item 1
          </mock-menuitem>
          <mock-menuitem
            classname="kebabMenuIcon"
            icon="[object Object]"
            text="item 2"
          >
            item 2
          </mock-menuitem>
        </mock-menulist>
      </mock-menupopover>
    </mock-menu>
    <mock-toast
      deps="[object Object]"
      timeoutlength="6000"
    />
  </div>
</DocumentFragment>
`;
